external/tools/modify_imgui_for_gearsvk.ps1

```diff
@@ -0,0 +1,292 @@
+# Modify imgui vulkan impl source file (imgui_impl_vulkan.cpp)
+#
+# This adds:
+# - User texture comment to the implemented features list at the top:
+#	//  [x] Renderer: User texture binding. ImTextureID is used to store a handle to a Descriptorset.
+#
+# - API call to get or create a texture
+#	IMGUI_IMPL_API ImTextureID  ImGui_ImplVulkan_Create_Or_GetTexture(VkSampler sampler, VkImageView image_view, VkImageLayout image_layout);
+#
+# - Add code for texture descriptor set binding after "vkCmdSetScissor\(command_buffer, 0, 1, &scissor\);" in ImGui_ImplVulkan_RenderDrawData:
+# 	// Bind texture descriptor set stored as ImTextureID'
+# 	VkDescriptorSet desc_set[1] = { (VkDescriptorSet) pcmd->TextureId };'
+# 	vkCmdBindDescriptorSets(command_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, g_PipelineLayout, 0, 1, desc_set, 0, NULL);'
+#
+# This changes:
+# - The comment "// Bind pipeline and descriptor sets:" in ImGui_ImplVulkan_SetupRenderState to:
+#	// Bind pipeline
+#
+# - The line "io.Fonts->SetTexID((ImTextureID)(intptr_t)g_FontImage);" in ImGui_ImplVulkan_CreateFontsTexture for setting the font texture id to:
+#	ImTextureID texID = (ImTextureID)g_DescriptorSet;
+#	io.Fonts->SetTexID(texID);
+#
+# This removes:
+# - User texture binding as missing feature comment at the top:
+#	//  [ ] Renderer: User texture binding
+#
+# - The unecessary font texture descriptor binding:
+#	VkDescriptorSet desc_set[1] = { g_DescriptorSet };
+#	vkCmdBindDescriptorSets(command_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, g_PipelineLayout, 0, 1, desc_set, 0, NULL);
+##
+
+"- Modifying " + $imgui_impl_vulkan_cpp + ":"
+
+# Read imgui vulkan impl source
+$fileContent = Get-Content $imgui_impl_vulkan_cpp
+
+# Helpers for function detection
+$inSetupRenderState = 0
+$inRenderDrawData = 0
+$inCreateFontsTexture = 0
+
+# Create an empty array and use it as the modified file
+$fileContentModified = @()
+
+# Header in file to communicate modification and prevent
+# multiple executions of this script
+$gearsVkHeader = "// Modified version for Gears-Vk"
+$fileContentModified += $gearsVkHeader
+$fileContentModified += '// Additions:'
+$fileContentModified += '// - User texture binding and descriptor set caching'
+$fileContentModified += ''
+
+$fileAlreadyModified = 0
+Foreach ($line in $fileContent)
+{
+	# Skip file if it already contains our header
+	if ($line -match $gearsVkHeader) {
+		Write-Host " -> already modified" -ForegroundColor Green
+		$fileAlreadyModified = 1
+		break
+	}
+
+	# Add user texture binding to implemented features list
+	if ($line -match "// Missing features:") {
+		$fileContentModified += "//  [x] Renderer: User texture binding. ImTextureID is used to store a handle to a Descriptorset."
+	}
+	# Remove user texture binding comment from missing feature
+	if ($line -match "//  \[ \] Renderer: User texture binding") {
+		continue
+	}
+
+	# Detect if in ImGui_ImplVulkan_SetupRenderState
+	if ($line -match "^static void ImGui_ImplVulkan_SetupRenderState\(") {
```

Comment:
Maybe protect the regex against changes in white spaces?! That should work with `\s+`, meaning "any whitespace, and at least one of it" (I think, at least).

Reply:
Maybe the `static` does not have to be included in the regex...then we would be safe if the `static` declaration was ever removed. Do you think, this could happen? Would the location still be uniquely identifiable without the `static`?